Controller/Adminhtml/Menu/Delete.php

```diff
@@ -58,25 +60,11 @@ public function __construct(
      */
     public function execute()
     {
-        $id = $this->getRequest()->getParam('id');
-
         try {
+            $id = $this->getRequestMenuId();
             $menu = $this->menuRepository->getById($id);
             $this->menuRepository->deleteById($id);
 
-            $filterBuilder = $this->filterBuilderFactory->create();
-            $filter = $filterBuilder->setField('menu_id')->setValue($id)->setConditionType('eq')->create();
-
-            $filterGroupBuilder = $this->filterGroupBuilderFactory->create();
-            $filterGroup = $filterGroupBuilder->addFilter($filter)->create();
-
-            $searchCriteriaBuilder = $this->searchCriteriaBuilderFactory->create();
-            $searchCriteria = $searchCriteriaBuilder->setFilterGroups([$filterGroup])->create();
-
-            $nodes = $this->nodeRepository->getList($searchCriteria);
-            foreach ($nodes->getItems() as $node) {
-                $this->nodeRepository->delete($node);
-            }
```

Comment:
As this block is removed, I think `filterBuilderFactory`, `filterGroupBuilderFactory`, `searchCriteriaBuilderFactory` and `nodeRepository` properties can be removed as well along with `use` statements and constructor assignments.

Reply:
✔️  Sorted